lib/js/src/manager/SystemCapabilityManager.js

```diff
@@ -31,11 +31,619 @@
+        const retrievedCapability = response.getSystemCapability()[getCapabilityMethodName]();
+        this.setCapability(systemCapabilityType, retrievedCapability);
+        return retrievedCapability;
+    }
+
+    /**
+     * @param {SystemCapabilityType} systemCapabilityType
+     * @param {function} listener
+    */
+    addOnSystemCapabilityListener (systemCapabilityType, listener) {
+        if (systemCapabilityType === null || systemCapabilityType === undefined
+            || listener === null || listener === undefined) {
+            return;
+        }
+        if (!this._onSystemCapabilityListeners[systemCapabilityType]) {
+            this._onSystemCapabilityListeners[systemCapabilityType] = [];
+        }
+        this._onSystemCapabilityListeners[systemCapabilityType].push(listener);
+        this.queryCapability(systemCapabilityType); // send out a query, since the caller wants an update for this capability
+    }
+
+    /**
+     * @param {SystemCapabilityType} systemCapabilityType
+     * @param {function} listener
+     * @return {Boolean}
+    */
+    removeOnSystemCapabilityListener (systemCapabilityType, listener) {
+        if (systemCapabilityType === null || systemCapabilityType === undefined
+            || listener === null || listener === undefined) {
+            return false;
+        }
+        const listenerArray = this._onSystemCapabilityListeners[systemCapabilityType];
+        if (!Array.isArray(listenerArray)) {
+            return false;
+        }
+        let removed = false;
+        // remove matching references to the passed in service listener
+        this._onSystemCapabilityListeners[systemCapabilityType] = listenerArray.filter(listenerCheck => {
+            if (listener === listenerCheck) {
+                removed = true;
+            }
+            return listener !== listenerCheck;
+        });
+        return removed;
+    }
+
+    /**
+     * Sets up listeners to pass events to the LifecycleManager
+    */
+    _setupRpcListeners () {
+        const listener = (message) => {
+            if (message !== null) {
+                if (message.getRPCType() === RpcType.RESPONSE) {
+                    switch (FunctionID.valueForKey(message.getFunctionName())) {
+                        case FunctionID.SetDisplayLayout: {
+                            this._displayCapabilities = message.getDisplayCapabilities();
+                            this._buttonCapabilities = message.getButtonCapabilities();
+                            this._presetBankCapabilities = message.getPresetBankCapabilities();
+                            this._softButtonCapabilities = message.getSoftButtonCapabilities();
+                            if (this._shouldConvertDeprecatedDisplayCapabilities) {
+                                this.setCapability(SystemCapabilityType.DISPLAYS, this._createDisplayCapabilityList(message));
+                            }
+                            break;
+                        }
```

Comment:
Bracket syntax for switch cases is not used commonly throughout the project, please remove :)